luaui/Widgets/cmd_blueprint.lua

```diff
@@ -272,6 +274,70 @@ local function setSelectedBlueprintIndex(index)
 	end
 end
 
+local function isValidBlueprint(blueprint)
+	if blueprint == nil then
+		return false
+	end
+
+	if blueprint.hasInvalidUnits then
+		return false
+	end
+
+	local buildable, unbuildable = WG["api_blueprint"].getBuildableUnits(blueprint)
+
+	if buildable == 0 then
+		return false
+	end
+
+	return true
+end
+
+local function getNextFilteredBlueprintIndex(startIndex, checkStart)
```

Comment:
Looks like checkStart is never used, so the initial isValidBlueprint and check, return etc before the for looks superfluous as well. Same thing for getPrevFilteredBlueprintIndex.

You're even just passing startIndex-1 when want to check the first index, so looks like don't need the initial test at all.

Reply:
Ah, good catch; I ended up refactoring use of that parameter out. I'll update it.